shared/engine/index.js

```diff
@@ -281,4 +279,12 @@ export type ResponseType = {
   error: (...args: Array<any>) => void,
 }
 
-export default engine
+let engine
+const makeEngine = () => {
```

Comment:
I think this is an improvement on what was before, but one drawback is that the timing of creating the engine is implicit based on first use. This approach introduces a global but defers the organizational task of deciding where to fit it into the namespace and when to initialize it. This is technical debt.

Do we have some other place where shared globals are setup / stored? Should we create a pattern for this?